### PR TITLE
Itinerary items create

### PR DESCRIPTION
**Pull request by James-Cochran**
- Created tests for creating itineraries with associated items
- Initial happy path tests are passing
- Still need sad path and edge case tests for all of the itineraries request spec

- Added protect_from_forgery and skip_before_action to the itineraries controller to get past the CSRF token authentication
- Added some conditional logic to associate created items with the creation of an itinerary
- I feel this needs some refactoring still, but it is up and passing rspec.
- Postman returns a 200 for a post with items and a rails c I can see the items.
- Still need to create a show action to see the specific itinerary.

### CODE REVIEW

File: app/controllers/api/v1/itineraries_controller.rb [hunk 1]
```diff
@@ -1,5 +1,39 @@
...
+  skip_before_action :verify_authenticity_token
+
   def index
     render json: ItinerarySerializer.format_itinerary_list(Itinerary.all), status: :ok
   end
+
+  def create
+    params[:itinerary][:user_id] = params[:itinerary][:user_id].to_i
+    itinerary = Itinerary.new(itinerary_params)
+    # binding.pry
```
Comment (Crosswolfv1) [inline app/controllers/api/v1/itineraries_controller.rb]: binding.pry here needs to be removed

File: app/controllers/api/v1/itineraries_controller.rb [hunk 2]
```diff
@@ -1,5 +1,39 @@
...
+    end
+  end
+
+  private
+  def itinerary_params
+    params.require(:itinerary).permit(:city, :duration, :user_id)
+  end
+
+  def item_params(item_data)
+    item_data.permit(:name, :address, :item_type, :phone, opening_hours: [])
```
Comment (Crosswolfv1) [inline app/controllers/api/v1/itineraries_controller.rb:36]: Theses are the correct datatypes AFAIK

**Review by Crosswolfv1 — CHANGES_REQUESTED** [review]

**Review by aloraalee — APPROVED** [review]
Great job with this and great job showing the multiple items in the request spec! It appears that this line on lines 93 and 94 of the itineraries_request_spec was repeated on lines 96 and 97.
itinerary = Itinerary.last
expect(itinerary.items.count).to eq(2)

Thanks for the through description in your PR! Super helpful!

**Comment by James-Cochran** [comment]
removed duplicate lines from test and straggler pry from the controller

**Review by Crosswolfv1 — APPROVED** [review]
changes made approved on this end
